source/dubregistry/web.d

```diff
@@ -556,6 +583,45 @@ class DubRegistryFullWebFrontend : DubRegistryWebFrontend {
 		redirect("/my_packages");
 	}
 
+	@auth @path("/my_packages/:packname/set_logo")
+	void postSetLogo(scope HTTPServerRequest request, string _packname, User _user)
+	{
+		enforceUserPackage(_user, _packname);
+		Path path;
+		foreach (key, file; request.files) {
+			if (key != "logo")
+				removeFile(file.tempPath);
+			else {
+				path = Path(tempDir()) ~ PathEntry(file.tempPath.toNativeString.baseName
+					~ file.filename.name.extension);
+				moveFile(file.tempPath, path);
+			}
+		}
+		enforceBadRequest(!path.empty);
+		m_registry.setPackageLogo(_packname, path);
+
+		redirect("/my_packages/"~_packname);
+	}
+
+	@auth @path("/my_packages/:packname/delete_logo")
+	void getDeleteLogo(scope HTTPServerRequest request, string _packname, User _user)
+	{
+		enforceUserPackage(_user, _packname);
+		m_registry.unsetPackageLogo(_packname);
```

Comment:
hm I'm not really a fan of deleting small files on a webserver. Especially as the logos are basically thumbnails (well if we drop webp it will just be pngs which are a bit bigger). I don't know about how much resources the dub registry has but idk about not keeping logos in case the user wants to restore them we can implement it now too.

Reply:
But that means we need to give them unique names (e.g. +timestamp( to make them distinguishable and chances are that everyone uploads 10 images before being pleased with the result. Even more for an admin there is hardly a way to purge unused icons.
I'd be fine with keeping the last 1 or 2 icons for undo buttons.
But building an icon gallery to pick one among all uploaded ones would be very out of scope.